astropy/io/fits/connect.py

```diff
@@ -32,6 +33,54 @@
 # Column-specific keywords regex
 COLUMN_KEYWORD_REGEXP = '(' + '|'.join(KEYWORD_NAMES) + ')[0-9]+'
 
+# TDISPn for both ASCII and Binary tables
+TDISP_RE_DICT = {}
+TDISP_RE_DICT['F'] = re.compile(r'(?:(?P<formatc>[F])(?:(?P<width>[0-9]+)\.{1}'
+                                r'(?P<precision>[0-9])+)+)|')
+TDISP_RE_DICT['A'] = TDISP_RE_DICT['L'] = \
+    re.compile(r'(?:(?P<formatc>[AL])(?P<width>[0-9]+)+)|')
+TDISP_RE_DICT['I'] = TDISP_RE_DICT['B'] = \
+    TDISP_RE_DICT['O'] = TDISP_RE_DICT['Z'] =  \
+    re.compile(r'(?:(?P<formatc>[IBOZ])(?:(?P<width>[0-9]+)'
+               r'(?:\.{0,1}(?P<precision>[0-9]+))?))|')
+TDISP_RE_DICT['E'] = TDISP_RE_DICT['G'] = \
+    TDISP_RE_DICT['D'] = \
+    re.compile(r'(?:(?P<formatc>[EGD])(?:(?P<width>[0-9]+)\.'
+               r'(?P<precision>[0-9]+))+)'
+               r'(?:E{0,1}(?P<exponential>[0-9]+)?)|')
+TDISP_RE_DICT['EN'] = TDISP_RE_DICT['ES'] = \
+    re.compile(r'(?:(?P<formatc>E[NS])(?:(?P<width>[0-9]+)\.{1}'
+               r'(?P<precision>[0-9])+)+)')
+
+# mapping from TDISP format to python format
+# A: Character
+# L: Logical (Boolean)
+# I: 16-bit Integer
+#    Can't predefine zero padding and space padding before hand without
+#    knowing the value being formatted, so grabbing precision and using that
+#    to zero pad, ignoring width. Same with B, O, and Z
+# B: Binary Integer
+# O: Octal Integer
+# Z: Hexadecimal Integer
+# F: Float (64-bit; fixed decimal notation)
+# EN: Float (engineering fortran format, exponential multiple of thee
+# ES: Float (scientific, same as EN but non-zero leading digit
+# E: Float, exponential notation
+#    Can't get exponential restriction to work without knowing value
+#    before hand, so just using width and precision, same with D, G, EN, and
+#    ES formats
+# D: Double-precision Floating Point with exponential
+#    (E but for double precision)
+# G: Double-precision Floating Point, may or may not show exponent
+TDISP_FMT_DICT = {'I' : '{{:{width}d}}',
+                  'B' : '{{:{width}b}}',
+                  'O' : '{{:{width}o}}',
+                  'Z' : '{{:{width}x}}',
+                  'F' : '{{:{width}.{precision}f}}',
+                  'G' : '{{:{width}.{precision}g}}'}
+TDISP_FMT_DICT['A'] = TDISP_FMT_DICT['L'] = '{{:>{width}}}'
+TDISP_FMT_DICT['E'] = TDISP_FMT_DICT['D'] =  \
+    TDISP_FMT_DICT['EN'] = TDISP_FMT_DICT['ES'] ='{{:{width}.{precision}e}}'
```

Comment:
I don't think all this should be in `fits/connect.py`. It should maybe go in `fits/column.py`, together with the ASCII equivalents ?

Reply:
@SaOgaz - I think all the TDISP related stuff (including the functions `_parse_tdisp_format` and `_fortran_to_python_format`) should be moved out of fits.connect, and fits.column seems the most logical place.

Reply:
Okay, will do.

Reply:
For the record, you agreed putting it in connect seemed reasonable in the linked other PR...

"@saraogaz - It seems a good plan. Conversion from FITS to Table and the opposite is indeed managed in connect.py, so it seems logical to convert the formats there. If we do not care about exactly round-trip for the few formats that do not have exact equivalents, then it is certainly the simplest way to proceed. This solution is perfectly fine for me, as you said above (I think most users wouldn't mind that these values were changed, as long as they were still sensible values)."

Reply:
Right, but it does not mean that all the implementation must be in connect.py. I think it is more logical to have the implementation detail in column.py, as it is quite similar to the TFORM code.